components/PinToHomescreen/index.tsx

```diff
@@ -0,0 +1,205 @@
+import React from "react"
+import { Image } from "react-bootstrap"
+
+import { getOS } from "../../lib/download"
+import { ACTIONS, ACTION_TYPE } from "../../pages/merchant/_reducer"
+import Modal from "../CustomModal/modal"
+import {
+  chromeModalContent,
+  desktopIosModalContent,
+  iosModalContent,
+  mobileChromeModalContent,
+} from "./browser-modal-content"
+import styles from "./pinToHomescreen.module.css"
+
+interface Props {
+  pinnedToHomeScreenModalVisible: boolean
+  dispatch: React.Dispatch<ACTION_TYPE>
+}
+
+const browser = (function () {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  const test = function (regexp) {
+    if (typeof window !== "undefined") {
+      return regexp.test(window.navigator.userAgent)
+    }
+  }
+  switch (true) {
+    case test(/edg/i):
+      return "Microsoft-Edge"
+    case test(/trident/i):
+      return "Microsoft-Internet-Explorer"
+    case test(/firefox|fxios/i):
+      return "Mozilla-Firefox"
+    case test(/opr\//i):
+      return "Opera"
+    case test(/ucbrowser/i):
+      return "UC-Browser"
+    case test(/samsungbrowser/i):
+      return "Samsung-Browser"
+    case test(/chrome|chromium|crios/i):
+      return "Google-Chrome"
+    case test(/safari/i):
+      return "Apple-Safari"
+    default:
+      return "Other"
+  }
+})()
+
+const PinToHomescreen = ({ pinnedToHomeScreenModalVisible, dispatch }: Props) => {
+  const os = getOS()
+  const [isOpen, setIsOpen] = React.useState<boolean>(pinnedToHomeScreenModalVisible)
```

Comment:
This state variable is not required, you can just use ```pinnedToHomeSceenModalVisible``` directly. This will avoid duplicating state and attempting to keep them in sync.

Reply:
I don't quite follow. I'll be needing the `setIsOpen` function to set the modal state. Or what am I not getting correctly here?

Reply:
You can just change line 71 to be ```isOpened={pinnedToHomeScreenModalVisible}``` and delete line 58

Reply:
Oh! I understand now